### PR TITLE
BUGFIX[29054]

### PR DESCRIPTION
**Pull request by iszmais**
https://mantis.ilias.de/view.php?id=29054

Dieser PR verbessert die Ansicht der Tag-Ergebnisse im Hauptmenü und steigert die User-Satisfactory.

Vorher:
image

Nachher:
image

### CODE REVIEW

File: Services/Tagging/classes/class.ilTaggingSlateContentMenuGUI.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            $this->ui->factory()->symbol()->glyph()->back(),
+            $this->lng->txt("back"), "#")->withOnLoadCode(function ($id) use ($tag_cmd) {
+            return "$(\"#$id\").click(function() { 
+                il.Util.ajaxReplaceInner('$tag_cmd', 'il-tag-slate-container'); return false;
+            });";
+        });
+
+        $items[] = $this->ui->factory()->item()->standard(sprintf(
+            $this->lng->txt("tagging_resources_for_tag"),
+            "<i>" . $_GET["tag"] . "</i>"
```
Comment (mjansenDatabay) [inline Services/Tagging/classes/class.ilTaggingSlateContentMenuGUI.php]: Please check if this is vulnerable against any XSS attacks. If so, please use `ilUtil::stripSlashes()`.

Comment (iszmais) [reply Services/Tagging/classes/class.ilTaggingSlateContentMenuGUI.php]: Thx, this was indeed not XSS safe.
Within fixing this i encountered that the tagging has problems with quotings in general (Tags with quote cant be resolved in the Menu). This fix does not fix this problems completly (tag will now be resolved but always with an empty result) but solves at least the security issues.

**Comment by alex40724** [comment]
I fixed the exit issue. Please present the presentation at the PLR II meetings.

**Comment by klees** [comment]
Hi @iszmais,

I hereby offer my help to find out how this could be discussed in the community.

Feel free to ask me anytime!

**Comment by iszmais** [comment]
Thx!

Due to some time issues i will not be able to look further into this before 4. annual quarter.
I will get in touch with you when in can find the time.

Greeting,

Ingmar

**Comment by chfsx** [comment]
Please excuse my late feedback! From my point of view, it makes sense to carry out this conversion. Visually it looks right, on the code side I would have the following application/wish:

Until now the concerned `Providers` delivered so called `Complex` items, which can contain any HTML. With your conversion, however, corresponding `Items` would have to be used, which are also intended for this purpose. For example, the link list can be used as a `LinkList` item, regular links as `Link` items. The possible `Items` from the GlobalScreen can be found here: src/GlobalScreen/Scope/MainMenu/Factory/MainMenuItemFactory.php

**Comment by kergomard** [comment]
@iszmais

We stumbled upon this PR in our TB meeting. Could you please take a look at this?

Thanks you very much,
Stephan

**Comment by Amstutz** [comment]
Hi @iszmais 

Thx for this proposal. I believe it to be a step in the right direction. However, there is no open related issue around this anymore. If you want to proceed further with this, shortly explain, why this is important from a UI or Technical perspective and why this does not need an FR. 

In my role as TB Member checking requests that are open for a long amount of time, I close this for the moment. You are welcome to re-open if you wish to continue the discussion here.

Thx.

**Pull request closed without merging**